full/src/main/java/apoc/ttl/TTLLifeCycle.java

```diff
@@ -45,22 +46,34 @@ public void start() {
             long ttlScheduleDb = configValues.schedule;
             ttlIndexJobHandle = scheduler.schedule(TTL_GROUP, this::createTTLIndex, (int)(ttlScheduleDb*0.8), TimeUnit.SECONDS);
             long limitDb = configValues.limit;
-            ttlJobHandle = scheduler.scheduleRecurring(TTL_GROUP, () -> expireNodes(limitDb), ttlScheduleDb, ttlScheduleDb, TimeUnit.SECONDS);
+            long batchSizeDb = configValues.batchSize;
+            ttlJobHandle = scheduler.scheduleRecurring(TTL_GROUP, () -> expireNodes(limitDb, batchSizeDb), ttlScheduleDb, ttlScheduleDb, TimeUnit.SECONDS);
         }
     }
 
-    public void expireNodes(long limit) {
+    public void expireNodes(long limit, long batchSize) {
         try {
             if (!Util.isWriteableInstance(db)) return;
-            db.executeTransactionally("MATCH (t:TTL) where t.ttl < timestamp() WITH t LIMIT $limit DETACH DELETE t",
-                    Util.map("limit", limit),
-                    result -> {
-                        QueryStatistics stats = result.getQueryStatistics();
-                        if (stats.getNodesDeleted()>0) {
-                            log.info("TTL: Expired %d nodes %d relationships", stats.getNodesDeleted(), stats.getRelationshipsDeleted());
-                        }
-                        return null;
-                    });
+            String withLimit = (limit > 0) ? "LIMIT $limit" : "";
+            String matchTTL = "MATCH (t:TTL) WHERE t.ttl < timestamp() ";
```

Comment:
let's just use `limit` as parameter for batch size, we don't need this inner limit anymore it was just used as a "poor mans" single batch in the beginning.

But now with periodic iterate we have proper batching and can just delete everything in batches of size "limit"